cpu/atmega_common/periph/rtt.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (C) 2016 Robert Hartung <hartung@ibr.cs.tu-bs.de>
+ *               2018 Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/*
+ * @ingroup cpu_atmega_common
+ * @ingroup drivers_periph_rtt
+ * @{
+ *
+ * @file
+ * @brief       Low-level ATmega RTT driver implementation
+ *
+ * @author      Robert Hartung <hartung@ibr.cs.tu-bs.de>
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * @}
+ */
+
+#include <avr/interrupt.h>
+
+#include "cpu.h"
+#include "irq.h"
+#include "periph/rtt.h"
+#include "periph_conf.h"
+#include "thread.h"
+
+#ifdef MODULE_XTIMER
+#include "xtimer.h"
+#endif
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* guard file in case no RTT device is defined */
+#if RTT_NUMOF
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if RTC_NUMOF
+extern void atmega_rtc_incr(void);
+#endif
+
+typedef struct {
+    uint8_t ext_cnt[3];      /* Counter to make 8-bit timer 32-bit */
```

Comment:
How about just using uint32_t? IMO the two wasted bytes in the struct might be worth increasing readability throughut the file.

Reply:
I started with uint32_t, but the fact that I was wasting two bytes (and increasing the memory usage by 20%) kept bothering me. It doesn't seem like a lot, but you could be using this on the ATmega328P, which only has 2kB of ram!

Reply:
These are now uint16_t, making the RTT a 24-bit timer. Both readable, and no wasted bytes.